File: 921. Minimum Add to Make Parentheses Valid/minimum_add_to_make_parentheses_valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        stack<int> stack;
        int count = 0;
        for (auto &ch : s) {
            if (ch == '(') {
                stack.push(ch);
                count++;
            } else {
                if (!stack.empty()) {
                    stack.pop();
                    count--;
                } else {
                    count++;
                }
            }
        }
        return count;
    }
};

File: 921. Minimum Add to Make Parentheses Valid/minimum_add_to_make_parentheses_valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        open = close = 0
        for c in s:
            if c == '(':
                open += 1
            else:
                if open > 0:
                    open -= 1
                else:
                    close += 1
        return open + close